src/transforms/RemoveFunctions.cpp

```diff
@@ -0,0 +1,115 @@
+#include "llvm/Pass.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/Analysis/InstructionSimplify.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Analysis/CallGraph.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/GlobalValue.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/Type.h"
+#include "llvm/IR/Operator.h"
+#include "llvm/Analysis/MemoryDependenceAnalysis.h"
+#include "llvm/IR/IntrinsicInst.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Transforms/Utils/SimplifyLibCalls.h"
+#include "llvm/Analysis/TargetLibraryInfo.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Dominators.h"
+#include "llvm/IR/ValueMap.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+#include "llvm/Analysis/CallGraph.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+
+#include <string.h>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+static llvm::cl::opt<std::string> function_list("functions_to_remove", llvm::cl::init(""));
+
+namespace previrt {
+namespace transforms {
+
+using namespace llvm;
+using namespace std;
+
+struct RemoveFunctions : public FunctionPass {
+	
+	static char ID;
+	RemoveFunctions() : FunctionPass(ID) {}
+        
+	bool runOnFunction(Function &F) override {
+
+		Module* module = F.getParent();
+		vector<string> removeFunctions;
+		stringstream string_to_stream(function_list);
+
+		while (string_to_stream.good()) {
+			string substr;
+        		getline(string_to_stream, substr, ',');
+        		removeFunctions.push_back(substr);
+    		}
+
+//		vector<string> removeFunctions{"func1","func2","func3","func4","add"};
+    
+		// Declare printf function
```

Comment:
Create a function `callPrintf` or something to help reading the code

Reply:
Done. I created `makePrintf` to follow the convention Rafae was following.